serverside-javascript-best-practices.md

```diff
@@ -320,6 +350,54 @@ createBooking.call(self, product, params, 'upgrades', 'supplier', 'payment', fun
 
 
 
+## Handling heavily sequential async code with promises
+
+* !! Using promises for heavily sequential async code has the downside of hiding the data being passed between functions. We lose the ability to wrap each layer in a closure to selectively pass the correct data between the correct entites. Everything must flow through the middle to get to the end. This breaks our rules on single-responsibility, but allows for slightly cleaner orchestration:
+* !! Sinon promises do not support sandboxing yet, so you'll have to manually restore everything by hand
+* !! It's impossible to prove that the data flow between promises is correct, only that data moves between them.
```

Comment:
I've chatted to @hxoliverrumbelow about this and we've talked about including examples in an appendix at the bottom of this document so the message isn't cluttered up with additional information.